generator/vertx/src/main/java/io/fabric8/maven/generator/vertx/VertxGenerator.java

```diff
@@ -16,32 +16,65 @@
 
 package io.fabric8.maven.generator.vertx;
 
-import java.util.List;
-
-import io.fabric8.maven.core.util.Configs;
+import io.fabric8.maven.core.util.MavenUtil;
 import io.fabric8.maven.docker.config.ImageConfiguration;
 import io.fabric8.maven.generator.api.GeneratorContext;
 import io.fabric8.maven.generator.javaexec.JavaExecGenerator;
-import io.fabric8.utils.Strings;
+import org.apache.maven.model.Plugin;
 import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.project.MavenProject;
+
+import java.util.List;
+import java.util.Map;
 
 /**
  * Vert.x Generator.
- *
+ * <p>
  * Then main part of creating the base image is taken java-exec generator.
- *
+ * <p>
  * In addition this generator here adds an appropriate healthcheck when vertx-dropwizard-metrics is
  * given as a dependency (TODO)
  */
 public class VertxGenerator extends JavaExecGenerator {
 
-    public VertxGenerator(GeneratorContext context) {
-        super(context, "vertx");
-    }
+  private static final String VERTX_MAVEN_PLUGIN_GA = "io.fabric8:vertx-maven-plugin";
+  private static final String SHADE_PLUGIN_GA = "org.apache.maven.plugins:maven-shade-plugin";
+  private static final String VERTX_GROUPID = "io.vertx";
+
+  public VertxGenerator(GeneratorContext context) {
+    super(context, "vertx");
+  }
+
+  @Override
+  public boolean isApplicable(List<ImageConfiguration> configs) throws MojoExecutionException {
+    return shouldAddImageConfiguration(configs)
+        && (MavenUtil.hasPlugin(getProject(), VERTX_MAVEN_PLUGIN_GA)
+        || MavenUtil.hasDependency(getProject(), VERTX_GROUPID));
+  }
+
+  @Override
+  protected Map<String, String> getEnv(boolean isPrePackagePhase) throws MojoExecutionException {
+    Map<String, String> ret = super.getEnv(isPrePackagePhase);
+    ret.put("JAVA_OPTIONS", "-Dvertx.cacheDirBase=/tmp");
```

Comment:
might be a good idea to examine whether the parent already added a `JAVA_OPTIONS` and possibly add this to an already existing entry.